src/NUnitFramework/framework/Attributes/TestCaseAttribute.cs

```diff
@@ -301,7 +303,10 @@ private TestCaseParameters GetParametersForTestCase(IMethodInfo method)
                             int length = argsProvided - argsNeeded + 1;
                             Array array = Array.CreateInstance(elementType, length);
                             for (int i = 0; i < length; i++)
-                                array.SetValue(parms.Arguments[argsNeeded + i - 1], i);
+                            {
+                                var argValue = ParamAttributeTypeConversions.Convert(parms.Arguments[argsNeeded + i - 1], elementType);
+                                array.SetValue(argValue, i);
+                            }
```

Comment:
Sorry, let me wrap my head back into this.
I "think" this was to support conversion of `string` -> `DateTimeOffset` if the receiving function expected an array of `DateTimeOffset`s, but you are absolutely correct that it would also apply to non-`DateTimeOffset` parameters too. Would this make more sense to split off into a separate issue/PR?

Reply:
Splitting it out sounds really helpful, if that’s ok with you! 😊

Reply:
Slowly chipping away at the PR feedback, but this piece is done. Changes are backed out.